dist-git/Dockerfile

```diff
@@ -0,0 +1,133 @@
+# dist-git Dockerfile
+# We cannot use a golang image as building the kubelet requires rsync and that is not present plus there is no easy way
+# to install it in the golang image.
+FROM openshift/golang-builder:rhel_8_1.15 as build
+LABEL stage=build
+RUN yum -y install rsync make go git tar findutils diffutils
+WORKDIR $REMOTE_SOURCE_DIR/build/
+
+# Build WMCO
+WORKDIR $REMOTE_SOURCE_DIR/build/windows-machine-config-operator
+# Copy .git metadata so that we can generate the version for the WMCO binary
+COPY $REMOTE_SOURCE/app/.git .git
+# Copy files and directories needed to build the WMCO binary
+COPY $REMOTE_SOURCE/app/build build
+COPY $REMOTE_SOURCE/app/cmd cmd
+COPY $REMOTE_SOURCE/app/deploy deploy
+COPY $REMOTE_SOURCE/app/hack hack
+COPY $REMOTE_SOURCE/app/pkg pkg
+COPY $REMOTE_SOURCE/app/test test
+COPY $REMOTE_SOURCE/app/vendor vendor
+COPY $REMOTE_SOURCE/app/version version
+COPY $REMOTE_SOURCE/app/go.mod go.mod
+COPY $REMOTE_SOURCE/app/go.sum go.sum
+COPY $REMOTE_SOURCE/app/Makefile Makefile
+COPY $REMOTE_SOURCE/app/tools.go tools.go
+COPY $REMOTE_SOURCE/app/bundle.Dockerfile bundle.Dockerfile
+RUN make build
+
+# Build WMCB
+WORKDIR $REMOTE_SOURCE_DIR/build/windows-machine-config-operator/windows-machine-config-bootstrapper/
+COPY $REMOTE_SOURCE/app/windows-machine-config-bootstrapper/ .
+RUN make build
+
+# Build hybrid-overlay
+WORKDIR $REMOTE_SOURCE_DIR/build/windows-machine-config-operator/ovn-kubernetes/
+COPY $REMOTE_SOURCE/app/ovn-kubernetes/ .
+WORKDIR $REMOTE_SOURCE_DIR/build/windows-machine-config-operator/ovn-kubernetes/go-controller/
+RUN make windows
+
+# Build Kubernetes node binaries
+WORKDIR $REMOTE_SOURCE_DIR/build/windows-machine-config-operator/kubernetes/
+COPY $REMOTE_SOURCE/app/kubernetes/ .
+ENV KUBE_BUILD_PLATFORMS windows/amd64
+RUN make WHAT=cmd/kubelet
+RUN make WHAT=cmd/kube-proxy
+
+# Build CNI plugins
+WORKDIR $REMOTE_SOURCE_DIR/build/windows-machine-config-operator/containernetworking-plugins/
+COPY $REMOTE_SOURCE/app/containernetworking-plugins/ .
+ENV CGO_ENABLED=0
+RUN ./build_windows.sh
+
+# Build the operator image with following payload structure
+# /payload/
+#├── cni
+#│   ├── flannel.exe
+#│   ├── host-local.exe
+#│   ├── win-bridge.exe
+#│   ├── win-overlay.exe
+#│   └── cni-conf-template.json
+#├── hybrid-overlay-node.exe
+#├── kube-node
+#│   ├── kubelet.exe
+#│   └── kube-proxy.exe
+#├── powershell
+#│   └── wget-ignore-cert.ps1
+#│   └── hns.psm1
+#└── wmcb.exe
+
+FROM openshift/golang-builder:rhel_8_1.15
```

Comment:
We should stick to the ubi minimal image here. Unlike CI we don't need go in the release image. I realize there will be a difference between CI and prod but we are going to have to live with that.